assets/src/edit-story/panels/shared.js

```diff
@@ -44,26 +44,30 @@ export const ActionButton = styled.button`
 	font-size: 11px;
 `;
 
-function InputGroup( { label, value, isMultiple, onChange, postfix, disabled } ) {
+function InputGroup( { type, label, value, isMultiple, onChange, postfix, disabled } ) {
 	const placeholder = isMultiple ? '( multiple )' : '';
+	const isCheckbox = type === 'checkbox';
 	return (
 		<Group disabled={ disabled }>
 			<Label>
 				{ label }
 			</Label>
 			<Input
+				type={ type || 'text' }
 				disabled={ disabled }
-				onChange={ ( evt ) => onChange( evt.target.value, evt ) }
+				onChange={ ( evt ) => onChange( isCheckbox ? evt.target.checked : evt.target.value, evt ) }
 				onBlur={ ( evt ) => evt.target.form.dispatchEvent( new window.Event( 'submit' ) ) }
 				placeholder={ placeholder }
-				value={ value }
+				value={ isCheckbox ? null : value }
+				checked={ isCheckbox ? value : null }
 			/>
 			{ postfix }
 		</Group>
 	);
 }
 
 InputGroup.propTypes = {
+	type: PropTypes.string,
```

Comment:
Doesn't this need a default value or React will start throwing warnings when missing? A default of `number` seems appropriate.

Reply:
Done